tests/test_mailchimp_discovery.py

```diff
@@ -0,0 +1,159 @@
+    """Test tap discover mode and metadata conforms to standards."""
+
+    def name(self):
+        """Returns name of the test"""
+        return "tap_tester_mailchimp_discover_test"
+
+    def test_run(self):
+        """
+        Testing that discovery creates the appropriate catalog with valid metadata.
+        • Verify number of actual streams discovered matches expected
+        • Verify the stream names discovered were what we expect
+        • Verify stream names follow the naming convention
+          streams should only have lowercase alphas and underscore
+        • Verify there is only 1 top level breadcrumb
+        • Verify there are no duplicate metadata entries
+        • Verify replication key(s)
+        • Verify primary key(s)
+        • Verify that if there is a replication key we are doing INCREMENTAL otherwise FULL
+        • Verify the actual replication matches our expected replication method
+        • Verify that primary, replication keys are given the inclusion of automatic.
+        • Verify that all other fields have the inclusion of available metadata.
+        """
+        streams_to_test = self.expected_check_streams()
+
+        conn_id = connections.ensure_connection(self, payload_hook=None)
+
+        # Verify that there are catalogs found
+        found_catalogs = self.run_and_verify_check_mode(conn_id)
+
+        # Verify stream names follow the naming convention
+        # Streams should only have lowercase alphas and underscores
+        found_catalog_names = {c["tap_stream_id"] for c in found_catalogs}
+        self.assertTrue(
+            all([re.fullmatch(r"[a-z_]+", name) for name in found_catalog_names]),
+            msg="One or more streams don't follow standard naming",
+        )
+
+        for stream in streams_to_test:
+            with self.subTest(stream=stream):
+
+                # Verify the catalog is found for a given stream
+                catalog = next(
+                    iter(
+                        [
+                            catalog
+                            for catalog in found_catalogs
+                            if catalog["stream_name"] == stream
+                        ]
+                    )
+                )
+                self.assertIsNotNone(catalog)
+
+                # Collecting expected values
+                expected_primary_keys = self.expected_primary_keys()[stream]
+                expected_replication_keys = self.expected_replication_keys()[stream]
+                expected_automatic_fields = self.expected_automatic_fields().get(stream)
+                expected_replication_method = self.expected_replication_method()[stream]
+
+                # Collecting actual values...
+                schema_and_metadata = menagerie.get_annotated_schema(
+                    conn_id, catalog["stream_id"]
+                )
+                metadata = schema_and_metadata["metadata"]
+
+                # BUG: TDL-20301 EMPTY BREADCRUMB IS NOT GENERATED
```

Comment:
This ticket has been resolved and in code review. Can we merge those changes and uncomment the assertion for test execution.

Reply:
done